scipy/stats/tests/test_stats.py

```diff
@@ -7949,50 +7949,41 @@ def test_no_args_gh20661(self):
 
 @array_api_compatible
 class TestCombinePvalues:
-
-    def test_fisher(self, xp):
-        # Example taken from https://en.wikipedia.org/wiki/Fisher%27s_exact_test#Example
-        xsq, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='fisher')
-        xp_assert_close(p, xp.asarray(0.02156), rtol=1e-4)
-
-    def test_stouffer(self, xp):
-        Z, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='stouffer')
-        xp_assert_close(p, xp.asarray(0.01651), rtol=1e-3)
-
-    def test_stouffer2(self, xp):
-        Z, p = stats.combine_pvalues(xp.asarray([.5, .5, .5]), method='stouffer')
-        xp_assert_close(p, xp.asarray(0.5), rtol=1e-4)
-
-    def test_weighted_stouffer(self, xp):
+    # Example taken from https://en.wikipedia.org/wiki/Fisher%27s_exact_test#Example
+    # Reference values computed using the following R code:
+    # > options(digits=16)
+    # > library(metap)
+    # > x1 = c(0.01, 0.2, 0.3)
+    # > x2 = c(0.5, 0.5, 0.5)
+    @pytest.mark.parametrize("method,pvalues,expected",
+                             # > sumlog(x1)
+                             [("fisher", [.01, .2, .3], 0.02156175132483465),
+                              # > sumz(x1)
+                              ("stouffer", [.01, .2, .3], 0.01651203260896294),
+                              # > sumz(x2)
+                              ("stouffer", [.5, .5, .5], 0.5),
+                              # need to work out how to generate this
+                              ("pearson", [.01, .2, .3], 0.02213),
```

Comment:
@mdhaber I got stuck finding another library that implements this, any suggestions?